docs/design/foreign_pod_tracking.md

```diff
@@ -0,0 +1,326 @@
+deprecation of this event. It will be removed in Yunikorn 1.8.
+We will also generate `NODE_ALLOC` for non-Yunikorn allocations, marking it foreign in the
+"message" field.
+
+|Event type|Event ID|State|When is it generated|
+|--|--|--|--|
+|Node|NODE_OCCUPIED|Existing - to be removed in 1.8|Occupied resource changes on a node|
+|Node|NODE_ALLOC|Already exists for Yunikorn allocations|Allocation (node assignment) occurs|
+
+**Constants**
+
+One constans will be introduced as a key for application tags. All pods scheduled by a
+different scheduler than YuniKorn will have the tag `foreign`.
+Values for the foreign tag are:
+- `static`: indicates a static pod, hard skip for preemption
+- `default`: preemptable pod
+
+### Scheduler core
+
+**Scheduling context**
+
+Processing updates happens in `ClusterContext.updateNode()`. There is a branch for
+`NodeInfo_UPDATE`:
+
+```
+case si.NodeInfo_UPDATE:
+  if sr := nodeInfo.SchedulableResource; sr != nil {
+    partition.updatePartitionResource(node.SetCapacity(resources.NewResourceFromProto(sr)))
+  }
+
+  if or := nodeInfo.OccupiedResource; or != nil {
+    node.SetOccupiedResource(resources.NewResourceFromProto(or))
+  }
+```
+
+Proposed changes: remove the path `or != nil`, because that field won’t be used.
+
+**Allocation object**
+
+The `objects.Allocations` type needs to be extended. The scheduler core needs to know if
+the allocation was scheduled by Yunikorn or not and if we want to support preemption, we
+have to mark allocations which represent static pods. These will be determined from the
+allocation tags explained in the K8s shim part.
+
+Two flags (boolean) to be added to the objects.Allocations that can only be set on
+creation. The flags can be read but never modified.
+
+**Partition**
+
+After YUNIKORN-2457, adding foreign allocations became much simpler. Some minor
+changes inside `PartitionContext.UpdateAllocation()` are necessary to ignore the
+application lookup for foreign pods.
+
+We might track foreign pods as part of an application in the future but that is out of scope of
+this change.
+
+
+**REST Interface**
+
+The way how allocations are counted gives the user more detailed information about the
+resource usage of all pods. Now we can show individual foreign allocations, not just a single
+counter.
+
+In Yunikorn 1.7, the data in the REST output will not change. We can still calculate
+OccupiedResources information per node. We also announce that AllocatedResources is
+about to change, because the value will be based on all pods on the node.
+
+From Yunikorn 1.8 onwards, we remove OccupiedResources and update the calculation of
+`AllocatedResources`.
+
+To support easier integration with the web UI, a separate list will be maintained for the two
+types of allocations. Allocations made by Yunikorn are still available under `allocations`.
+Non-Yunikorn allocations will be listed under `foreign_allocations`.
```

Comment:
Besides allocations, we're also tracing container count metrics too(allocated, released, rejected)(YuniKorn only). Should we also trace non-YuniKorn container count? 

In the front page of Web UI, we called below API to get totalContainers count:
- http://localhost:9889/ws/v1/history/containers

The totalContainers was calculated by "totalContainersRunning := allocatedContainers(YuniKorn) - releasedContainers(YuniKorn)

Reply:
> Should we also trace non-YuniKorn container count?

After rethink it, I believe we don't need to trace it.
Knowning the total non-YuniKorn container count doesn't help us much. 
I only want to know how many running containers are scheduled by YuniKorn in all applications.